internal/gh/gh.go

```diff
@@ -26,12 +28,31 @@ func CreatePullRequest(args ...string) error {
 	return nil
 }
 
-func ensureGHInstalled() error {
+func EnsureInstalledAndAuthorized() error {
 	cmd := exec.Command("which", "gh")
```

Comment:
Consider:

```
cmd := exec.Command("command", "-v", "gh")
```

[Apparently](https://stackoverflow.com/a/677212), `command -v` is more reliable as `which` can have weird behaviour, including not setting an exit status. This would cause the following lines to not work correctly.